pkg/cmd/reset.go

```diff
@@ -100,5 +105,43 @@ func runReset(opts *resetOpts) error {
 
 	s.Logger.Warnln("this command will require an explicit confirmation starting with the next minor release (v1.3)")
 
+	fmt.Println("The following nodes will be resetted: ")
+	for _, node := range s.Cluster.ControlPlane.Hosts {
+		fmt.Printf("\t+ reset control plane node %q (%s)\n", node.Hostname, node.PrivateAddress)
+	}
+	for _, node := range s.Cluster.StaticWorkers.Hosts {
+		fmt.Printf("\t+ reset static worker nodes %q (%s)\n", node.Hostname, node.PrivateAddress)
+	}
+
+	if !s.Cluster.MachineController.Deploy {
+		s.Logger.Info("Skipping deleting workers because machine-controller is disabled in configuration.")
+	}
+
+	if s.DynamicClient == nil {
+		return errors.New("kubernetes client not initialized")
+	}
+
+	// Gather information about machine-controller managed nodes
+	ctx := context.Background()
+	nodes := &corev1.NodeList{}
+	if err := s.DynamicClient.List(ctx, nodes); err != nil {
+		return errors.Wrap(err, "unable to list nodes")
+	}
+
+	for _, node := range nodes.Items {
+		fmt.Printf("\t+ reset machine-controller managed nodes %q\n", node.Name)
+	}
```

Comment:
There are two problems with this approach:

1. This will also list nodes not managed by machine-controller (e.g. static worker nodes)
2. Someone might run `kubeone reset` on instances that are not provisioned yet. In that case, reset will fail, however, that shouldn't happen at all.

The first point can be solved by filtering by labels or by listing Machine objects (instead of Node objects). The second point can be solved by ignoring errors instead of returning them.

Reply:
@xmudrii  does this resolve this issue? I wasn't sure if v1alpha1.MachineList{} is here correct :)
https://github.com/kubermatic/kubeone/pull/1251/commits/cc83fc094ccbc6f93c11135832fe715f28a5cb0b